Unix/protocol/protocol.c

```diff
@@ -490,8 +615,16 @@ static MI_Boolean _SendAuthRequest(
         }
     }
 
-    req->uid = geteuid();
-    req->gid = getegid();
+    if (uid != INVALID_ID && gid != INVALID_ID)
+    {
+        req->uid = uid;
+        req->gid = gid;
+    }
+    else
+    {
+        req->uid = geteuid();
```

Comment:
which process are we in here? Why would we get invalid ids here? it does not seem to make sense to do this